### ISSUE TITLE
Cannot operate on list of tuple literals

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1097]
**Describe the bug**
Tuple literals contained in mutable sequences are interpreted as containing `Literal`s.  When attempting to add values of the same type to the sequence, or when attempting to pass the sequence as an argument to a function, Pyright produces a variety of general type errors.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That none of the operations below would produce an error.

**Screenshots or Code**
```python
# pyright: strict

foo = [(1, 1), (2, 2)]
foo.append((3, 3))      # error
foo.extend([(4, 4)])    # error
foo + [(5, 5)]          # error

bar = [*foo, (3, 3)]

def baz(value: list[tuple[int, int]]): ...

baz(bar)                # error
```

**VS Code extension or command-line**
extension, 1.1.78

**Comment by erictraut** [comment]
This is intended behavior. Refer to [this documentation](https://github.com/microsoft/pyright/blob/master/docs/type-inference.md#tuple-expressions). If the inferred type is not to your liking, you can provide an explicit type declaration for the variable.

```python
foo: List[Tuple[int, int]] = [(1, 1), (2, 2)]
```

**Comment by layday** [comment]
I wasn't sure if that behaviour should extend to tuple literals contained in mutable sequences, though I suppose providing an annotation is a small price to pay.
